frontend/src/components/common/formQuestions/AddQuestionModal.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useState } from "react";
+
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  VStack,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  Select,
+  Input,
+  FormErrorMessage,
+  ModalFooter,
+  HStack,
+  Button,
+  Checkbox,
+  Box,
+  IconButton,
+  Radio,
+} from "@chakra-ui/react";
+import { AddIcon, DeleteIcon } from "@chakra-ui/icons";
+import {
+  CreateFormQuestion,
+  QuestionCategory,
+  QuestionType,
+} from "../../../types/CampsTypes";
+
+const QuestionOptionSection = ({
+  questionType,
+  setOptionsArray,
+}: {
+  questionType: string;
+  setOptionsArray: React.Dispatch<React.SetStateAction<string[]>>;
+}): React.ReactElement => {
+  interface Option {
+    option: string;
+  }
+
+  const key: keyof Option = "option";
+
+  const [options, setOptions] = useState<Array<Option>>([{ option: "" }]);
+
```

Comment:
Alternatively, why can't we just use an array?

Reply:
I did this because of the dynamic form in the add modal (the user should be able to add and remove option fields in the form). I couldn't get it working to begin with so I followed a tutorial that did it this way lol. I am kinda scared to go back cuz it took me a while 😅

Reply:
LOL I mean if it works it works ig